### PR TITLE
Task6 - Test your REST services

### PR DESCRIPTION
**Pull request by Irenkia**
1)    Remove all the excluded classes from Gradle config. The only class which may remain excluded is InvoiceApplication, it’s not intended to be run by tests.
2)    Increase the required coverage to 85% on both lines and branch level.
3)    Add a basic test using @Stepwise - it will be very similar to the one presented in the lesson.
4)    Add exhaustive tests to your controller - it should handle all corner cases (e.g. invoice not found, deleting one out of multiple invoices).

It’s forbidden to use @Stepwise in this test, each test case should be independent - e.g. add 3 invoices, remove 2nd, verify there are only 2 left.
Please make sure to create helper methods - e.g. to add invoice to the system.
Measure the code coverage with IntelliJ running only this test - all methods in InvoiceController should be covered, Jacoco will measure global coverage - for development purpose you need something more selective - IntelliJ is perfect for this task.

Make sure tests are running correctly on both database implementations - memory & file based. Be aware that InFileDatabase may store data between test runs so it’s wise to make sure the database is empty before starting the test. Do not assume any ids - always use ids returned by POST method.

Please make sure the application is building, running & tests are passing.
Checkstyle and code coverage should pass without warnings. All should build with Gradle & Github Actions.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: CI
+
+on:
+  push:
+    branches: [task*]
+  pull_request:
```
Comment (PiotrFarbaniec) [inline .github/workflows/main.yml:6]: Jeśli nazwy gałęzi masz ustawione na lesson** , to będzie konflikt tutaj??

File: build.gradle [hunk 2]
```diff
@@ -125,14 +125,14 @@ jacocoTestCoverageVerification {
...
-                minimum = 0.50
+                minimum = 0.85
             }
         }
 
         rule {
             limit {
                 counter = 'LINE'
-                minimum = 0.50
+                minimum = 0.85
```
Comment (PiotrFarbaniec) [inline build.gradle:135]: Super testy, mucha nie siada :)

**Review by PiotrFarbaniec — APPROVED** [review]
OK

File: .github/workflows/main.yml (hunk 1 above)
Comment (Irenkia) [reply .github/workflows/main.yml:6]: Nazywam gałęzie „task”

Comment (Irenkia) [reply .github/workflows/main.yml:6]: nie będzie żadnego konfliktu

**Review by marwis1995 — APPROVED** [review]
